### DISCUSSION TITLE
Trying to use cosmocc with CMake project

### DISCUSSION THREAD

**Discussion by trickster** [jart/cosmopolitan#1320] [Q&A]
I came across [this](https://jcbhmr.me/2024/07/22/cosmocc-cmake/) post, this works perfectly fine on linux but doesn't work on Mac ARM. It constantly gives me error that it cannot compile simple test program in C while running `cmake -B build`.

**Comment by aquawicket** [comment]
Does CMake mention any variables not set, like CMAKE_C_COMPILER, CMAKE_CXX_COMPILER or CMAKE_MAKE_PROGRAM?   I know if you set some of those before you call PROJECT(), it will help cmake find your compiler.  Not sure how CMake treats cosmocc, so may wanna set them all yourself. What does CMake choose for CMAKE_GENERATOR. Is is Unix Makefiles?

**Reply by shemeshg** [reply]
I’ve created an example project that works on Linux, macOS, and Windows. You can find it here: 

No cosmopolitan:
https://github.com/shemeshg/movavg

Here’s the Cosmopolitan project with a readme that explains everything: https://github.com/shemeshg/movavg/tree/cosmopolitan_build.

However, I’m having a bit of trouble linking the static library of Cosmopolitan to the main project. As a result, it’s all in one project for now.



Let me know if you have any questions!

**Comment by Tridu33** [comment]
https://github.com/madebr/cosmopolitan-cmake-toolchain/blob/master/test/CMakeLists.txt
